test/v1alpha1/git_checkout_test.go

```diff
@@ -78,6 +78,61 @@ func TestGitPipelineRun(t *testing.T) {
 	}
 }
 
+// Test revision fetching with refspec specified
+func TestGitPipelineRunWithRefspec(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		description string
+		revision    string
+		refspec     string
+	}{{
+		description: "Fetch refs/tags/v0.1.0 alongside master and checkout the master branch",
+		revision:    "master",
+		refspec:     "refs/tags/v0.1.0:refs/tags/v0.1.0 refs/heads/master:refs/heads/master",
+	}, {
+		description: "Checkout specific revision from refs/pull/1009/head's commit chain",
+		revision:    "968d5d37a61bfb85426c885dc1090c1cc4b33436",
+		refspec:     "refs/pull/1009/head",
+	}, {
+		description: "Fetch refs/pull/1009/head into a named master branch and then check it out",
+		revision:    "master",
+		refspec:     "refs/pull/1009/head:refs/heads/master",
+	}} {
+		t.Run(tc.description, func(t *testing.T) {
+			c, namespace := setup(t)
+			knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+			defer tearDown(t, c, namespace)
+
+			t.Logf("Creating Git PipelineResource %s", gitSourceResourceName)
```

Comment:
I think it makes sense to log in the failure case with the `t.Fatalf` but these kinds of status lines can make test logs quite noisy to trace through when looking at verbose output.  I suggest removing the use of `t.Logf` unless absolutely necessary.  What do you think?

Reply:
I've removed the log entries in the latest squashed commit.